cypress/e2e/swap.test.ts

```diff
@@ -81,35 +77,33 @@ describe('Swap', () => {
   it('can swap ETH for USDC', () => {
     const TOKEN_ADDRESS = USDC_MAINNET.address
     const BALANCE_INCREMENT = 1
-    cy.visit('/swap', { ethereum: 'hardhat' })
-      .then((window) => {
-        hardhat = window.hardhat
-      })
-      .then(() => hardhat.utils.getBalance(hardhat.wallet.address, USDC_MAINNET))
-      .then((balance) => Number(balance.toFixed(1)))
-      .then((initialBalance) => {
-        cy.get('#swap-currency-output .open-currency-select-button').click()
-        cy.get(getTestSelector('token-search-input')).clear().type(TOKEN_ADDRESS)
-        cy.contains('USDC').click()
-        cy.get('#swap-currency-output .token-amount-input').clear().type(BALANCE_INCREMENT.toString())
-        cy.get('#swap-currency-input .token-amount-input').should('not.equal', '')
-        cy.get('#swap-button').click()
-        cy.get('#confirm-swap-or-send').click()
-        cy.get(getTestSelector('dismiss-tx-confirmation')).click()
-
-        cy.then(() => hardhat.send('hardhat_mine', ['0x1', '0xc'])).then(() => {
-          // ui check
-          cy.get('#swap-currency-output [data-testid="balance-text"]').should(
-            'have.text',
-            `Balance: ${initialBalance + BALANCE_INCREMENT}`
-          )
-
-          // chain state check
-          cy.then(() => hardhat.utils.getBalance(hardhat.wallet.address, USDC_MAINNET))
-            .then((balance) => Number(balance.toFixed(1)))
-            .should('eq', initialBalance + BALANCE_INCREMENT)
+    cy.hardhat().then((hardhat) => {
+      cy.then(() => hardhat.getBalance(hardhat.wallet.address, USDC_MAINNET))
+        .then((balance) => Number(balance.toFixed(1)))
+        .then((initialBalance) => {
+          cy.get('#swap-currency-output .open-currency-select-button').click()
+          cy.get(getTestSelector('token-search-input')).clear().type(TOKEN_ADDRESS)
+          cy.contains('USDC').click()
+          cy.get('#swap-currency-output .token-amount-input').clear().type(BALANCE_INCREMENT.toString())
+          cy.get('#swap-currency-input .token-amount-input').should('not.equal', '')
+          cy.get('#swap-button').click()
+          cy.get('#confirm-swap-or-send').click()
+          cy.get(getTestSelector('dismiss-tx-confirmation')).click()
+
+          cy.then(() => hardhat.provider.send('hardhat_mine', ['0x1', '0xc'])).then(() => {
```

Comment:
why did you set the interval value here?

Reply:
This is legacy - it's on our TODO list to move this to a hardhat function. The interval is required to explicitly run over deadline, I think.